pkg/sql/parser/parse_test.go

```diff
@@ -1042,7 +1042,7 @@ SELECT 0x FROM t
 		},
 		{
 			`SELECT x'fail' FROM t`,
```

Comment:
I'd also like to see an error case with the valid characters but wrong number of characters: `SELECT x'01a'`